### PR TITLE
feat(Exception-Handler): add basic exception handler which logs uncau…

### PR DESCRIPTION
**Pull request by nicoprow**
…ght exceptions

### CODE REVIEW

File: bpdm-common/src/main/kotlin/org/eclipse/tractusx/bpdm/common/exception/ResponseEntityExceptionHandler.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.eclipse.tractusx.bpdm.gate.exception
```
Comment (pazepaze) [inline bpdm-common/src/main/kotlin/org/eclipse/tractusx/bpdm/common/exception/ResponseEntityExceptionHandler.kt:20]: package does not match file location

File: bpdm-common/src/main/kotlin/org/eclipse/tractusx/bpdm/common/exception/ResponseEntityExceptionHandler.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        ex: Exception, request: WebRequest
+    ): ResponseEntity<Any>? {
+        val annotations = ex::class.findAnnotations(ResponseStatus::class)
+
+        if (annotations.isEmpty() || annotations.any { it.value == HttpStatus.INTERNAL_SERVER_ERROR }) {
+            //Prints with stack trace
+            kotlinLogger.error(ex) { "Caught internal server error" }
+        } else {
+            //Prints without stack trace
+            logger.debug(ex)
```
Comment (pazepaze) [inline bpdm-common/src/main/kotlin/org/eclipse/tractusx/bpdm/common/exception/ResponseEntityExceptionHandler.kt:49]: I assume that using another logger there was done on purpose?

**Review by pazepaze — CHANGES_REQUESTED** [review]

File: bpdm-common/src/main/kotlin/org/eclipse/tractusx/bpdm/common/exception/ResponseEntityExceptionHandler.kt (hunk 2 above)
Comment (nicoprow) [reply bpdm-common/src/main/kotlin/org/eclipse/tractusx/bpdm/common/exception/ResponseEntityExceptionHandler.kt:49]: Yep, this logger's behaviour to print an exception with message but without stack trace was quite handy. So I took the default exception handling logger here.

File: bpdm-common/src/main/kotlin/org/eclipse/tractusx/bpdm/common/exception/ResponseEntityExceptionHandler.kt (hunk 1 above)
Comment (nicoprow) [reply bpdm-common/src/main/kotlin/org/eclipse/tractusx/bpdm/common/exception/ResponseEntityExceptionHandler.kt]: Fixed

**Review by pazepaze — APPROVED** [review]

**Pull request merged**